pkg/registry/helpers.go

```diff
@@ -108,3 +112,68 @@ func GetRegistryName(imageName string) (string, error) {
 	}
 	return tag.Registry.Name(), nil
 }
+
+// checkForProxyConfigAndUpdateCert checks if proxy should be used to interact with registry host, if so, checks if user has configured CA cert data
+// for proxy host in the configuration file (using "tanzu cert config add --hostname <proxy-host> --ca-certificate /path/to/cacert) and use it
+func checkForProxyConfigAndUpdateCert(registryHost, scheme string, registryCertOpts *CertOptions) error {
+	proxyURL, err := getProxyURL(registryHost, scheme)
+	if err != nil {
+		return errors.Wrap(err, "checking proxy configuration")
+	}
+	// If there is no proxy, nothing to update
+	if proxyURL == nil {
+		return nil
+	}
+
+	// check if user provided cert configuration for proxy host, if so, use only the CACertData
+	if exists, _ := configlib.CertExists(proxyURL.Host); !exists {
+		return nil
+	}
+	// get cert provided for the proxy
+	cert, err := configlib.GetCert(proxyURL.Host)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get the custom certificate configuration for host %q", registryHost)
+	}
```

Comment:
Is it possible to combine these 2 calls into 1 `GetCert` call and `return nil` if an error occurs?